WordPress/theme-experiments
# Twenty Nineteen as a Block-based Theme

This PR uses a similar approach to #2, but applies it to Twenty Nineteen. It's based on the demo content from the [Twenty Nineteen theme demo](https://2019.wordpress.net), but implemented as a [Block-based Theme](https://developer.wordpress.org/block-editor/developers/themes/block-based-themes/). CSS is mostly just duplicated from Twenty Nineteen, but has a few small adjustments as noted in the code comments.

## Basic steps for testing: 

1. Create a fresh WordPress site
2. Clone this repository or upload a zip into the site’s themes directory
3. Install the Gutenberg plugin
4. Enable the Full Site Editing and Full Site Editing Demo Templates experiments in the `Gutenberg > Experiments` panel.
5. Activate the "Twenty Nineteen Blocks" theme from within `Appearance > Themes`. 
6. Visit the Customizer and hit "Publish"

You can compare to the [Twenty Nineteen Demo site](https://2019.wordpress.net). It should look pretty similar. 

## Issues 

This is still in very much in progress, so this is surely not an exhaustive list of issues:

- The menu doesn't work right if the starter content creates pages using IDs in their URLs instead of slugs.  
- The Cover Images at the top of pages don't exactly match the demo. This'll require a bunch of extra styles, so may not be worth it for this demo. 
- The site tagline/image don't correspond to the actual tagline/image.
- The site image doesn't hang off the left side like it does in the original theme.
- The menu does not collapse nicely on mobile. 
- ~~Button styles don't match.~~
- The blog page is currently empty. 

## Screenshots: 

![Screen Shot 2020-01-10 at 13 52 03](https://user-images.githubusercontent.com/1202812/72178428-dc35de80-33b0-11ea-8e71-ea47eaaa03b1.png)

![Screen Shot 2020-01-10 at 13 53 53](https://user-images.githubusercontent.com/1202812/72178444-e061fc00-33b0-11ea-84c9-17ee594606ae.png)

![Screen Shot 2020-01-10 at 13 54 01](https://user-images.githubusercontent.com/1202812/72178462-e657dd00-33b0-11ea-937f-21efe01d96ee.png)
